src/Stratis.Bitcoin.Features.LightWallet/LightWalletSyncManager.cs

```diff
@@ -247,34 +255,56 @@ public void SyncFromDate(DateTime date)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(date), date);
 
+            // Set the syncWalletDate to the date passed in parameter
+            if (this.syncWalletDate == DateTime.MinValue || this.walletTip.Header.BlockTime.LocalDateTime > date)
+            {
+                // This will update the condition within the async loop below
+                this.logger.LogTrace("Setting new wallet sync from date to {0}; wallet's previous sync from date was {1}.", date, this.syncWalletDate);
+                this.syncWalletDate = date;
+            }
+            else
+            {
+                this.logger.LogTrace("Ignoring request to sync from date {0} as the wallet is already syncing from older date {1}. Subsequent call to SyncFromHeight might amend the block hight.", date, this.walletTip.Header.BlockTime.LocalDateTime);
+            }
+
+            // Check if the task is already running and don't start another asyncloop call if we are already running one.
+            if (this.syncFromDateAsyncLoop != null)
+                return;
+
             // Before we start syncing we need to make sure that the chain is at a certain level.
             // If the chain is behind the date from which we want to sync, we wait for it to catch up, and then we start syncing.
             // If the chain is already past the date we want to sync from, we don't wait, even though the chain might not be fully downloaded.
-            if (this.chain.Tip.Header.BlockTime.LocalDateTime < date)
+            if (this.chain.Tip.Header.BlockTime.LocalDateTime < this.syncWalletDate)
             {
-                this.logger.LogTrace("The chain tip's date ({0}) is behind the date from which we want to sync ({1}). Waiting for the chain to catch up.", this.chain.Tip.Header.BlockTime.LocalDateTime, date);
+                this.logger.LogTrace("The chain tip's date ({0}) is behind the date from which we want to sync ({1}). Waiting for the chain to catch up.", this.chain.Tip.Header.BlockTime.LocalDateTime, this.syncWalletDate);
```

Comment:
Could this be more explicit? Maybe "The local bitcoin chain's highest block ({0}) is behind ...
" We're running 2 chains, not sure chain tip alone is clear